pages/nft-book-store/collection/status/[collectionId].vue

```diff
@@ -977,9 +1051,57 @@ function addNotificationEmail () {
   notificationEmailInput.value = ''
 }
 
-function onStripeConnectWalletInput () {
-  // force stripeConnectWallet to update when stripeConnectWalletInput is updated
-  stripeConnectWallet.value = stripeConnectWalletInput.value.trim()
+async function onStripeConnectWalletInput (input: any) {
+  if (!isStripeConnectChecked.value) { return }
+  const inputValue = input.target.value.trim()
+  stripeConnectWalletInput.value = inputValue
+  stripeConnectInputError.value = ''
+
+  if (!LIKE_ADDRESS_REGEX.test(inputValue)) {
+    stripeConnectInputError.value = 'You have entered an invalid wallet address'
+    return
+  }
+  if (stripeConnectStatusByWalletMap.value[inputValue]) {
+    return stripeConnectStatusByWalletMap.value[inputValue]
+  }
+  isStripeConnectLoading.value = true
+  stripeConnectStatusByWalletMap.value[inputValue] = { isReady: false }
+  try {
+    const { data: stripeConnectStatus, error } = await useFetch(
+      `${LIKE_CO_API}/likernft/book/user/connect/status?wallet=${inputValue}`
+    )
+    if (
+      error?.value &&
+      error?.value?.statusCode !== 404
```

Comment:
What if it's 404? Also this is the third time we have this code snippet, maybe change it to component or composable